### PR TITLE
Add .gitignore for new projects

### PR DESCRIPTION
**Pull request by adrhumphreys**
Added git ignore, to ignore the node_modules and dist folder for new projects

### What are you trying to accomplish with this PR?
This adds a gitignore file by default so that node_modules and dist don't get commited to sc by accident

[#126](https://github.com/Shopify/slate/issues/126)

### CODE REVIEW

File: src/.gitignore [hunk 1]
```diff
@@ -0,0 +1,4 @@
+# Custom #
+###################
+node_modules
+dist
```
Comment (chrisberthe) [inline src/.gitignore]: Can you add a trailing slash to these to signify they're directories?

**Comment by chrisberthe** [comment]
Thanks @adrhumphreys for the PR - just a small comment. @NathanPJF I've also assigned you for some 👀.

File: src/.gitignore (hunk 1 above)
Comment (adrhumphreys) [inline src/.gitignore]: Woops, added them in there 👍

**Comment by adrhumphreys** [comment]
Awesome, @chrisberthe I've add the trailing slashes

**Comment by t-kelly** [comment]
Hey @adrhumphreys 👋 - have you had a chance to test this out? The `.gitignore` file you've added won't do anything because the `dist/` and `node_modules/` don't live inside the `src/` folder, they live inside the root folder. We have a `.gitignore` file for the root directory that ignores both `node_modules/` and `dist/`.

Is the existing setup not working for you? Care to elaborate as to how you ended up with `dist/` and `node_modules/` in the `src/` folder?

**Comment by t-kelly** [comment]
Looks like the issue is that `.gitignore` is not being copied over when we create a new theme.

**Comment by chrisberthe** [comment]
As @t-kelly mentioned, this doesn't solve the issue of ignoring the `node_modules` and `dist` folder - the `.gitignore` should be placed inside of the root theme directory, not `src`.

Closing this PR for now, but we'll definitely revisit this soon - maybe we could initialize this with a flag upon running `slate theme` and/or `slate migrate`... but don't necessarily want to clutter up options for each of those commands either. We'll 🤔.

**Pull request closed without merging**